File: Example Code/01 Variables, Data Types, Input and Output/CookieRecipe3.java
import java.util.Scanner;

// Chocolate Chip cookie ingredients from:
// http://allrecipes.com/recipe/10813/best-chocolate-chip-cookies/

public class CookieRecipe3
{
	public static void main( String args[] )
	{
        // Math - allow user to input a ratio. Adjust the ingredient list quantities.
        Scanner scanner = new Scanner( System.in );
        double cupsOfButter = 1.0, cupsOfWhiteSugar = 1.0,
            cupsOfBrownSugar = 1.0, cupsOfFlour = 3.0,
            cupsOfWalnuts = 1.0, cupsOfChocolateChips = 2.0, eggs = 2.0,
            teaspoonsOfVanilla = 2.0, teaspoonsOfBakingSoda = 1.0,
            teaspoonsOfWater = 2.0, teaspoonsOfSalt = 0.5;
            
        System.out.println( "How many servings do you want to make?" );
        System.out.print( "Enter a ratio, can have decimal: " );
        double ratio = scanner.nextDouble();
        
        System.out.println( "Chocolate Chip Cookies" );
        System.out.println( cupsOfButter * ratio + " cups butter, softened " );
        System.out.println( cupsOfWhiteSugar * ratio + " cups white sugar" );
        System.out.println( cupsOfBrownSugar * ratio + " cups packed brown sugar" );
        System.out.println( eggs * ratio + " eggs" );
        System.out.println( teaspoonsOfVanilla * ratio + " teaspoons vanilla extract" );
        System.out.println( cupsOfFlour * ratio + " cups all-purpose flour" );
        System.out.println( teaspoonsOfBakingSoda * ratio + " teaspoon baking soda" );
        System.out.println( teaspoonsOfWater * ratio + " teaspoons hot water" );
        System.out.println( teaspoonsOfSalt * ratio + " teaspoons salt" );
        System.out.println( cupsOfChocolateChips * ratio + " cups semisweet chocolate chips" );
        System.out.println( cupsOfWalnuts * ratio + " cups chopped walnuts" );
    }
}

File: Example Code/01 Variables, Data Types, Input and Output/source/CookieRecipe3.cpp
#include <iostream>
using namespace std;

int main()
{
    // Ask user the ratio
    cout << "How many servings do you want to make? (1 for normal, or other value to adjust): ";
    float ratio;
    cin >> ratio;
    
    // Store amount for each
    float cupsOfButter = 1.0;
    float cupsOfWhiteSugar = 1.0;
    float cupsOfBrownSugar = 1.0;
    float cupsOfFlour = 3.0;
    float cupsOfWalnuts = 1.0;
    float cupsOfChocolateChips = 2.0;
    float eggs = 2.0;
    float teaspoonsOfVanilla = 2.0;
    float teaspoonsOfBakingSoda = 1.0;
    float teaspoonsOfWater = 2.0;
    float teaspoonsOfSalt = 0.5;
    
    // Output recipe information 
    cout << "Chocolate Chip Cookies" << endl;
    cout << cupsOfButter * ratio << " cups butter, softened " << endl;
    cout << cupsOfWhiteSugar * ratio << " cups white sugar" << endl;
    cout << cupsOfBrownSugar * ratio << " cups packed brown sugar" << endl;
    cout << eggs * ratio << " eggs" << endl;
    cout << teaspoonsOfVanilla * ratio << " teaspoons vanilla extract" << endl;
    cout << cupsOfFlour * ratio << " cups all-purpose flour" << endl;
    cout << teaspoonsOfBakingSoda * ratio << " teaspoon baking soda" << endl;
    cout << teaspoonsOfWater * ratio << " teaspoons hot water" << endl;
    cout << teaspoonsOfSalt * ratio << " teaspoons salt" << endl;
    cout << cupsOfChocolateChips * ratio << " cups semisweet chocolate chips" << endl;
    cout << cupsOfWalnuts * ratio << " cups chopped walnuts" << endl;
    
    return 0;
}
